### PR TITLE
HTML5 inputs

### PR DESCRIPTION
**Pull request by ricn**
I added the remaining HTML5 input types.

### CODE REVIEW

File: lib/phoenix/html/form.ex [hunk 1]
```diff
@@ -260,6 +260,69 @@ defmodule Phoenix.HTML.Form do
...
+    generic_input(:time, form, field, opts)
+  end
+
+  @doc """
+  Generates a datetime input.
+
+  See `text_input/3` for example and docs.
+  """
+  def datetime_input(form, field, opts \\ []) do
+    generic_input(String.to_atom("datetime-local"), form, field, opts)
```
Comment (josevalim) [inline lib/phoenix/html/form.ex]: `:"datetime-local"` :)

**Comment by josevalim** [comment]
Thank you. I will ready carefully about those and cherry pick the relevant commits in case we won't merge all.

**Comment by josevalim** [comment]
I have added the range_input only. All others are still not fully supported by major browsers and require special recommendations before using it according to http://www.wufoo.com/html5/ so I don't think we should promote them yet. Thank you!

**Comment by ricn** [comment]
Aha ok. I just added all of them because Rails supports them. I was looking here: http://guides.rubyonrails.org/form_helpers.html (chapter 1.4)

**Pull request closed without merging**